### DISCUSSION TITLE
Need help getting an external ControlNet script from A1111 SD to work in SD.Next

### DISCUSSION THREAD

**Discussion by ArchAngelAries** [vladmandic/sdnext#1317] [Q&A]
I'm not good at coding, so I need help. A friend (who is unavailable/away/unreachable at the moment) from discord wrote me a script to process images through ControlNet to generate ControlNet maps and bypassing the need for image generation. I prefer this method because it works in batch and doesn't require all images in a folder to be the same dimensions as it bypasses the Txt2Img & Img2Img height & width settings when running a batch through ControlNet in the normal way.

It worked in A1111 SD directml version until the recent update broke my 7900 XT's memory usage, but I learned SD.Next is way better for AMD.  If anyone can help me figure it out I'd greatly appreciate it... I have "--allow-code" enabled in the webui, and controlnet settings "Allow detectmap auto saving" and "Allow other script to control this extension" enabled in the settings. In A1111 SD I had to enable "--API" in the webui-user.bat, but I don't know how to (or if I need to) enable it in SD.Next... So, I'm not sure if that's why when I launch the script nothing happens or if the script needs to be tweaked to specifically work in SD.Next...

**Comment by ArchAngelAries** [comment]
Here's the code to the py script  

[autoCNet.txt](https://github.com/vladmandic/automatic/files/11649931/autoCNet.txt)

**Comment by vladmandic** [comment]
>  In A1111 SD I had to enable "--API" in the webui-user.bat, but I don't know how to (or if I need to) enable it in SD.Next

its always enabled here, you don't need to set it.
otherwise, this script uses a well-defined api - and if there are errors, you'd see something either in console where script is running or in server console.
if it "does nothing", its not finding any images to process to start with.
